packages/actor-query-source-identify-rdfjs/lib/ActorQuerySourceIdentifyRdfJs.ts

```diff
@@ -36,17 +37,32 @@ export class ActorQuerySourceIdentifyRdfJs extends ActorQuerySourceIdentify {
 
   public async run(action: IActionQuerySourceIdentify): Promise<IActorQuerySourceIdentifyOutput> {
     const dataFactory: ComunicaDataFactory = action.context.getSafe(KeysInitQuery.dataFactory);
+
+    // Convert action.querySourceUnidentified.value to RDF.Source
+    // If it's an RDF.Source, always prefer that interface over RDF.DatasetCore
+    const value = action.querySourceUnidentified.value;
+    const source: RDF.Source = typeof value === 'object' && this.isASource(value) ?
+      <RDF.Source> value :
+      new DatasetSourceWrapper(<RDF.DatasetCore> value);
+
     return {
       querySource: {
         source: new QuerySourceRdfJs(
-          <RDF.Source> action.querySourceUnidentified.value,
+          source,
           dataFactory,
           await BindingsFactory.create(this.mediatorMergeBindingsContext, action.context, dataFactory),
         ),
         context: action.querySourceUnidentified.context ?? new ActionContext(),
       },
     };
   }
+
+  private isASource(obj: RDF.Source | RDF.Store | RDF.DatasetCore): obj is RDF.Source | RDF.Store {
+    const result = obj.match();
```

Comment:
This call could be very expensive, as it essentially fetches all triples from the store/dataset.

I would recommend looking for the presence of certain members instead, such as `has`, `delete`, and `add`. https://rdf.js.org/dataset-spec/#datasetcore-interface